syntax = "proto3";

package protos;

// ping请求
message Ping {
	string message = 1;
}

// ping回应
message Pong {
	string message = 1;
}

// 连接认证请求
message LoginAuth {
	int64 code = 1; // 认证代码
	int32 id = 2; // 服务器ID
	int32 group = 3; // 服务器分组
	string name = 4; // 服务器名字
	int32 listen_client_port = 5; // 监听客户端端口,不监听填0
}

// 连接认证回应
message LoginAuthResult {
	int32 error = 1; // 0:成功,非0时失败
	string errmsg = 2; // 错误信息
}

// 服务参数
message ServiceParam {
	int64 p1 = 1; // 参数1
	int64 p2 = 2; // 参数2
	int64 p3 = 3; // 参数3
}

// 注册服务请求
message RegisterService {
	map<string, ServiceParam> services = 1; // k:服务名称 v:服务参数
}

// 注册服务回应
message RegisterServiceResult {
	int32 error = 1; // 0:成功,非0时失败
	string errmsg = 2; // 错误信息
}

// 注销服务请求
message UnregisterService {
	repeated string service_names = 1; // k:服务名称
}

// 注销服务回应
message UnregisterServiceResult {
	int32 error = 1; // 0:成功,非0时失败
	string errmsg = 2; // 错误信息
}

// 服务器信息
message ServerInfo {
	int32 id = 1; // 服务器ID
	int32 group = 2; // 服务器分组
	string name = 3; // 服务器名字
}

// 查询服务器信息请求
message QueryServerInfo {
	string name = 1; // 查询的服务器名字
}

// 查询服务器信息回应
message QueryServerInfoResult {
	repeated ServerInfo infos = 1; // 服务器信息
}